### PR TITLE
Instrument directory is read only in the install

### PR DESCRIPTION
**Pull request by peterfpeterson**
The `/opt/Mantid/instrument` directory being `rwx` on linux has been a source of issues at ORNL. Since `vtp` files are now written into the user's home area, this directory no longer needs to be world writable. It is important that this be installed (and uninstalled) on all supported platforms to make loading data still works. The tester should check that all have been verified. Please put your name next to the platform tested.
- [x] OSX - @quantumsteve
- [x] RHEL6 - @rosswhitfield
- [x] RHEL7 - @rosswhitfield
- [x] Ubuntu - @rosswhitfield
- [x] Windows - @FedeMPouzols 

**This must be tried on an installed version.** Start mantid twice: Once after you have deleted your `{appdata}/instrument` directory, then a second time to see that it works if that directory exists.

### CODE REVIEW

File: Framework/CMakeLists.txt [hunk 1]
```diff
@@ -152,10 +152,7 @@ add_custom_target( Framework DEPENDS ${FRAMEWORK_LIBS} )
 
 # Create instrument directory and make it writable so vtp files can go there
```
Comment (rosswhitfield) [inline Framework/CMakeLists.txt]: Need to update comment.

**Comment by rosswhitfield** [comment]
Testing on ubuntu I get the error
```
DownloadInstrument-[Error] Error in execution of algorithm DownloadInstrument:
DownloadInstrument-[Error] Access to file denied
```

**Comment by peterfpeterson** [comment]
@rosswhitfield I'm glad to know that somebody has a system that has it broken. Please list your value (in installation area and home area) for `instrumentDefinition.directory` and `instrumentDefinition.vtpDirectory`. Those seem to be the missing link, but there may be more `properties` at play.

**Comment by rosswhitfield** [comment]
I was testing this on `precise-buidler.sns.gov`. I cleared my home setting before testing it so I don't have any `instrumentDefinition` defined in my home config. But in `/opt/mantidunstable/bin/Mantid.properties` there is `instrumentDefinition.directory = ../instrument`

**Comment by rosswhitfield** [comment]
Installed the RHEL7 unstable build on `corelli` and get the same error with DownloadInstrument. You can try it yourself on there.
`/opt/mantidunstable/instrument` is set to `755`.

**Comment by rosswhitfield** [comment]
Found the problem, if `~/.mantid/instrument` folder doesn't exist at startup DownloadInstrument will write to `/opt/Mantid/instrument` changing the permissions on `Facilities.xml` and other files. I tested this with Mantid 3.5.

**Comment by peterfpeterson** [comment]
When `DownloadInstrument` runs it also writes a file `/opt/Mantid/instrument/github.json` that is `644` `${USER}:users`.

**Comment by rosswhitfield** [comment]
:+1: All working correctly now. Tested RHEL7 rpm.

**Comment by FedeMPouzols** [comment]
:+1: on my windoze 7.

**Comment by rosswhitfield** [comment]
:+1: RHEL6

**Comment by rosswhitfield** [comment]
Just a note, if I remove write permissions (`-w`) to the `/.mantid/instrument` folder I get a not very useful error.

```
An unhandled exception has been caught. MantidPlot will have to close. Details:

Access to file denied.
```

**Comment by quantumsteve** [comment]
:+1: on OS X.

**Comment by rosswhitfield** [comment]
:+1: Ubuntu

**Pull request merged**